src/device-state/current-state.ts

```diff
@@ -108,25 +108,25 @@ const getStateDiff = (): DeviceStatus => {
 	}
 
 	const diff = {
-		local: _(stateForReport.local)
-			.omitBy((val, key: keyof DeviceStatus['local']) =>
-				_.isEqual(lastReportedLocal[key], val),
-			)
-			.omit(INTERNAL_STATE_KEYS)
-			.value(),
-		dependent: _(stateForReport.dependent)
-			.omitBy((val, key: keyof DeviceStatus['dependent']) =>
-				_.isEqual(lastReportedDependent[key], val),
-			)
-			.omit(INTERNAL_STATE_KEYS)
-			.value(),
+		local: _.omitBy(
+			stateForReport.local,
+			(val, key: keyof NonNullable<DeviceStatus['local']>) =>
+				_.isEqual(lastReportedLocal[key], val) ||
+				INTERNAL_STATE_KEYS.includes(key) ||
+				!sysInfo.isSignificantChange(
+					key,
+					lastReportedLocal[key] as number,
+					val as number,
+				),
+		),
+		dependent: _.omitBy(
+			stateForReport.dependent,
+			(val, key: keyof DeviceStatus['dependent']) =>
+				_.isEqual(lastReportedDependent[key], val) ||
+				INTERNAL_STATE_KEYS.includes(key),
+		),
 	};
 
-	const toOmit: string[] = sysInfo.filterNonSignificantChanges(
-		lastReportedLocal as sysInfo.SystemInfo,
-		stateForReport.local as sysInfo.SystemInfo,
-	);
-	diff.local = _.omit(diff.local, toOmit);
```

Comment:
Same story, combining this in reduces the work/extra object and the corresponding cpu savings